seatunnel-connectors-v2/connector-amazondynamodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/amazondynamodb/sink/DdynamoDbSinkClient.java

```diff
@@ -0,0 +1,134 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class DdynamoDbSinkClient {
+    private final AmazondynamodbSourceOptions amazondynamodbSourceOptions;
+    private ScheduledExecutorService scheduler;
+    private ScheduledFuture<?> scheduledFuture;
+    private volatile boolean initialize;
+    private volatile Exception flushException;
+    private DynamoDbClient dynamoDbClient;
+    private final List<WriteRequest> batchList;
+    protected SeaTunnelRowDeserializer seaTunnelRowDeserializer;
+
+    public DdynamoDbSinkClient(AmazondynamodbSourceOptions amazondynamodbSourceOptions, SeaTunnelRowType typeInfo) {
+        this.amazondynamodbSourceOptions = amazondynamodbSourceOptions;
+        this.batchList = new ArrayList<>();
+        this.seaTunnelRowDeserializer = new DefaultSeaTunnelRowDeserializer(typeInfo);
+    }
+
+    private void tryInit() throws IOException {
+        if (initialize) {
+            return;
+        }
+        dynamoDbClient = DynamoDbClient.builder()
+            .endpointOverride(URI.create(amazondynamodbSourceOptions.getUrl()))
+            // The region is meaningless for local DynamoDb but required for client builder validation
+            .region(Region.of(amazondynamodbSourceOptions.getRegion()))
+            .credentialsProvider(StaticCredentialsProvider.create(
+                AwsBasicCredentials.create(amazondynamodbSourceOptions.getAccessKeyId(), amazondynamodbSourceOptions.getSecretAccessKey())))
+            .build();
+
+        scheduler = Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder().setNameFormat("DdynamoDb-sink-output-%s").build());
+        scheduledFuture = scheduler.scheduleAtFixedRate(
+            () -> {
+                try {
+                    flush();
+                } catch (IOException e) {
+                    flushException = e;
+                }
+            },
+            amazondynamodbSourceOptions.getBatchIntervalMs(),
+            amazondynamodbSourceOptions.getBatchIntervalMs(),
+            TimeUnit.MILLISECONDS);
+
+        initialize = true;
+    }
+
+    public synchronized void write(PutItemRequest putItemRequest) throws IOException {
+        tryInit();
+        checkFlushException();
+        batchList.add(WriteRequest.builder().putRequest(
+            PutRequest.builder().item(putItemRequest.item()).build()).build());
+        if (amazondynamodbSourceOptions.getBatchSize() > 0
+            && batchList.size() >= amazondynamodbSourceOptions.getBatchSize()) {
+            flush();
+        }
+    }
+
+    public synchronized void close() throws IOException {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+            scheduler.shutdown();
+        }
+        flush();
```

Comment:
The close method will also be called when no data is written, and dynamoDbClient will have a null pointer exception at this time. This can be tested by a task where the source does not output any data.